kit/Kit.cpp

```diff
@@ -2893,12 +2899,20 @@ bool anyInputCallback(void* data)
 
     if (queue->getTileQueueSize() > 0)
     {
+        bool tilesInsideVisibleArea = queue->anyOf([&document](const TileDesc& desc) {
+            return document->isTileDescInsideVisibleArea(desc);
+        });
+        // Pending tiles requests inside visible area -> allow writer layout (querying
+        // if it should quit to allow alternative work to run) to continue.
+        if (tilesInsideVisibleArea)
+            return false;
+        // Quit writer layout to serve tiles.
         return true;
```

Comment:
The logic with this change would be: if we have tiles to be rendered in the queue, then we tell Writer layout to continue, otherwise we interrupt. Don't we want to do the opposite? As in: if we have tiles to be rendered in the queue and those would update the visible area, then interrupt (so we can render those tiles), but otherwise go on (instead of the old behavior of always interrupting).

Reply:
yeah, but it seems that on rendering that (pending because interrupted?) layout work gets done and resize due to that don't get seen at the browser side